merlion/utils/autosarima_utils.py

```diff
@@ -247,12 +247,10 @@ def multiperiodicity_detection(x, pval=0.05, max_lag=None):
     # select the local maximum points with acf > 0
     candidates = np.intersect1d(np.where(xacf > 0), argrelmax(xacf)[0])
 
-    # the periods should be smaller than one third of the lenght of time series
-    candidates = candidates[candidates < int(x.shape[0] / 3)]
+    # the periods should be smaller than one half of the length of time series
+    candidates = candidates[candidates < int(x.shape[0] / 2)]
     if candidates.shape[0] == 0:
         return []
-    else:
-        candidates = candidates[np.insert(argrelmax(xacf[candidates])[0], 0, 0)]
```

Comment:
What was the rationale for removing the peak-finding code?

Reply:
`argrelmax` ignores both the start point and last point. Therefore, if we want to keep this strategy, we need to add the last point. This is the reason for the failure case of sales data you shared with me. In another viewpoint, we have incorporated the multi periodicity selection strategy, it can also help filter the periodicity.

I have inverted the implementation and add the process for last point